From 012631e24db4e27a151faede9b6750cfc87e4ca6 Mon Sep 17 00:00:00 2001
From: Andrey Borzenkov <arvidjaar@gmail.com>
Date: Wed, 27 Feb 2013 10:02:39 +0100
Subject: [PATCH 160/364] 	* grub-core/partmap/msdos.c
 (grub_partition_msdos_iterate): 	Fix off by one error in enumerating
 extended partitions.

---
 ChangeLog                 | 5 +++++
 grub-core/partmap/msdos.c | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index d770758..c77a6a6 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2012-02-27  Andrey Borzenkov <arvidjaar@gmail.com>
+
+	* grub-core/partmap/msdos.c (grub_partition_msdos_iterate):
+	Fix off by one error in enumerating extended partitions.
+
 2013-02-26  Andrey Borzenkov <arvidjaar@gmail.com>
 
 	* grub-core/disk/efi/efidisk.c(grub_efidisk_get_device_name): Fix
diff --git a/grub-core/partmap/msdos.c b/grub-core/partmap/msdos.c
index b0e11c4..0d0a6b7 100644
--- a/grub-core/partmap/msdos.c
+++ b/grub-core/partmap/msdos.c
@@ -196,7 +196,7 @@ grub_partition_msdos_iterate (grub_disk_t disk,
 	      if (hook (disk, &p, hook_data))
 		return grub_errno;
 	    }
-	  else if (p.number < 4)
+	  else if (p.number < 3)
 	    /* If this partition is a logical one, shouldn't increase the
 	       partition number.  */
 	    p.number++;
-- 
1.8.1.4

